src/TConsole.cpp

```diff
@@ -2875,3 +2887,70 @@ void TConsole::setProfileName(const QString& newName)
         pC->setProfileName(newName);
     }
 }
+
+std::pair<bool, QString> TConsole::closeWindow(const QString& name, const bool destroy)
+{
+    if (name.isEmpty()) {
+        return std::make_pair(false, QStringLiteral("a window/buffer cannot have an empty string as a name"));
```

Comment:
I _think_ the following might also work now that we're on C++17, if you'd like to try it:

```suggestion
        return {false, QStringLiteral("a window/buffer cannot have an empty string as a name")};
```

Reply:
Yeah, I think you are right.  Will do it in another commit with anything else that needs tweaking...

Reply:
Done in next commit.